### ISSUE TITLE
docutils.conf is not re-read on subsequent builds for alternate docs

### ISSUE THREAD

**Issue by plastikos** [sphinx-doc/sphinx#5358]
Subject: docutils.conf is not re-read on subsequent builds for alternate docs

### Problem
- `docutils.conf` will not be re-read when building for subsequent doc types - only the first built doc type.  If there is a setting in `docutils.conf` and `html` docs are built the setting will be applied.  If `singlehtml` is built next the settings will be ignored.  Once the environment is pickled the `docutils.conf` settings will not be used for subsequent builds that re-use the pickled environment.

#### Procedure to reproduce the problem
contents of `docutils.conf`:
```
[html4css1 writer]
field_name_limit = 40
option_limit = 40
```

```
$> rm -rf build
$> sphinx-build -M html "source" "build"
$> sphinx-build -M singlehtml "source" "build"
```

In this case the html docs will utilize the `field_name_limit` of `40` but the singlehtml docs will have a `field_name_limit` of the default, `14`.  This is true for any setting that might be put in `docutils.conf`.  Try the following and build two types of documentation:

`docutils.conf`:
```
[general]
dump_settings = 1
dump_internals = 1
```
```
$> rm -rf build
$> sphinx-build -M html "source" "build"
$> sphinx-build -M singlehtml "source" "build"
```
Dumping only happens with the first build of html.

#### Error logs / results
No errors

#### Expected results
Settings in `docutils.conf` should always be applied to all documentation builds whether the documentation type was built first or last in a series of documentation.

### Reproducible project / your project
[foo.zip](https://github.com/sphinx-doc/sphinx/files/2321980/foo.zip)
Referenced is a minimal test case that changes the `field_name_limit` setting.  Perform the following:
```
>$ tar xf foo.tar.gz
>$ cd foo
>$ make clean
>$ make html
>$ make singlehtml
```
Observe that the `html` docs utilize the larger `field_name_limit` setting but the `singlehtml` docs do not.  Do the following:
```
>$ make clean
>$ make singlehtml
>$ make html
```
Now the `singlehtml` docs utilize the larger `field_name_limit` setting but the `html` docs do not.

### Environment info
- OS: Debian Linux
- Python version: 3.6.6
- Sphinx version: 1.7.6

**Comment by tk0miya** [comment]
Fortunately, this is already fixed at 1.8. Could you check 1.8.0b1?

**Comment by tk0miya** [comment]
I'm closing this now because it seems already fixed.
Please feel free to reopen if you still get same error in 1.8.0b1.
Thanks,
